### PR TITLE
[TRAFFIC-9346][TRAFFIC-9347] Add confluent network network-link endpoint create/delete

### PR DESCRIPTION
**Pull request by ay-zhang**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli-az. We will merge commits into main from this feature branch after all of the commands for network are implemented.

- Add `confluent network network-link endpoint create`
- Add `confluent network network-link endpoint delete`

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests.

### CODE REVIEW

File: internal/network/command_network_link_endpoint_delete.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/resource"
+	"github.com/confluentinc/cli/v3/pkg/utils"
+)
+
+func (c *command) newNetworkLinkEndpointDeleteCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "delete <id>",
+		Short:             "Delete a network link endpoint.",
```
Comment (sgagniere) [inline internal/network/command_network_link_endpoint_delete.go:19]:
```suggestion
		Use:               "delete <id-1> [id-2] ... [id-n]",
		Short:             "Delete one or more network link endpoints.",
```

File: test/fixtures/output/network/network-link/endpoint/delete-help.golden [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Delete a network link endpoint.
+
+Usage:
+  confluent network network-link endpoint delete <id> [flags]
```
Comment (sgagniere) [inline test/fixtures/output/network/network-link/endpoint/delete-help.golden:4]:
```suggestion
Delete one or more network link endpoints.

Usage:
  confluent network network-link endpoint delete <id-1> [id-2] ... [id-n] [flags]
```

**Review by sgagniere — COMMENTED** [review]
some delete specific changes to match the style of our other descriptions:

File: internal/network/command_network_link_endpoint_create.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+	networkingv1 "github.com/confluentinc/ccloud-sdk-go-v2/networking/v1"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+)
+
+func (c *command) newNetworkLinkEndpointCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <name>",
```
Comment (syoung-cflt) [inline internal/network/command_network_link_endpoint_create.go:14]: `display_name` is optional in the [API spec](https://docs.confluent.io/cloud/current/api.html#tag/Network-Link-Endpoints-(networkingv1)/operation/createNetworkingV1NetworkLinkEndpoint).

I recently made an update to other sub-commands in https://github.com/confluentinc/cli/pull/2451 to make the `name` arg as optional. Please take a look at this.

You would need to update the `network-link service create` command as well. Feel free to do it in another PR.

Comment (ay-zhang) [reply internal/network/command_network_link_endpoint_create.go]: Updated to optional endpoint display name. 

The backend for `network-link service create` actually requires a display name, will update this once I change backend to match API spec.

**Review by sgagniere — APPROVED** [review]

**Review by syoung-cflt — APPROVED** [review]

**Pull request merged**